<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>900</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1">
     <widget class="QGraphicsView" name="graphicsView">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuSpeed_Detector">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_Image"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuOperations">
    <property name="title">
     <string>Operations</string>
    </property>
    <addaction name="actionEliminate_Colors"/>
    <addaction name="actionEdges"/>
    <addaction name="actionHarris_Corners"/>
    <addaction name="actionBlur"/>
    <addaction name="actionShowAngles"/>
    <addaction name="actionEdge_Thinning"/>
   </widget>
   <addaction name="menuSpeed_Detector"/>
   <addaction name="menuOperations"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionLoad_Image"/>
   <addaction name="actionReset"/>
   <addaction name="actionTrain"/>
   <addaction name="actionTrain_Harris"/>
   <addaction name="actionDetect"/>
   <addaction name="actionDetect_Harris"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="messagesDock">
   <property name="features">
    <set>QDockWidget::NoDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Messages</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0">
      <widget class="QPlainTextEdit" name="messagesTextEdit">
       <property name="undoRedoEnabled">
        <bool>false</bool>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="textInteractionFlags">
        <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByMouse</set>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QPlainTextEdit" name="timingsTextEdit">
       <property name="undoRedoEnabled">
        <bool>false</bool>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="textInteractionFlags">
        <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByMouse</set>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="features">
    <set>QDockWidget::NoDockWidgetFeatures</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="1" column="0" alignment="Qt::AlignTop">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Edge Threshold</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0" alignment="Qt::AlignTop">
      <widget class="QLineEdit" name="harrisThreshold">
       <property name="text">
        <string>10000000</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0" alignment="Qt::AlignTop">
      <widget class="QLineEdit" name="edgeThreshold">
       <property name="text">
        <string>50</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0" alignment="Qt::AlignTop">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Harris threshold</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionLoad_Image">
   <property name="text">
    <string>Load Image...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Reset</string>
   </property>
   <property name="toolTip">
    <string>Reset image</string>
   </property>
  </action>
  <action name="actionBlur">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Blur</string>
   </property>
   <property name="toolTip">
    <string>Add blur</string>
   </property>
  </action>
  <action name="actionEdges">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Edges</string>
   </property>
   <property name="toolTip">
    <string>Edge detection</string>
   </property>
  </action>
  <action name="actionShowAngles">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Show Angles</string>
   </property>
   <property name="toolTip">
    <string>Show sobel angles</string>
   </property>
  </action>
  <action name="actionEdge_Thinning">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Edge Thinning</string>
   </property>
   <property name="toolTip">
    <string>Apply a pass of edge thinning</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionTrain">
   <property name="text">
    <string>Train</string>
   </property>
  </action>
  <action name="actionEliminate_Colors">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Eliminate Colors</string>
   </property>
  </action>
  <action name="actionDetect">
   <property name="text">
    <string>Detect</string>
   </property>
  </action>
  <action name="actionHarris_Corners">
   <property name="text">
    <string>Harris Corners</string>
   </property>
  </action>
  <action name="actionTrain_Harris">
   <property name="text">
    <string>Train Harris</string>
   </property>
  </action>
  <action name="actionDetect_Harris">
   <property name="text">
    <string>Detect Harris</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
